### PR TITLE
Bump gradle-plugins to 0.4 (and fix one more thing in ReferencesPrinter)

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

File: muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/ReferencesPrinter.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
    */
   public static void printMuzzleReferences() {
     for (InstrumentationModule instrumentationModule :
-        ServiceLoader.load(InstrumentationModule.class)) {
+        ServiceLoader.load(InstrumentationModule.class, ReferencesPrinter.class.getClassLoader())) {
```
Comment (iNikem) [inline muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/ReferencesPrinter.java:29]: Why does this differ from the current context classloader?

Comment (mateuszrzeszutek) [reply muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/ReferencesPrinter.java:29]: That's what surprised me too a bit - apparently the current context CL is still the gradle CL; while we want to use the instrumentation CL for everything that happens in this class.

Comment (iNikem) [reply muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/ReferencesPrinter.java:29]: Should we do similar to https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/92a69c33097c7fc0d3d2453205c2f44c1289fbda/gradle-plugins/src/main/kotlin/io.opentelemetry.instrumentation.muzzle-check.gradle.kts#L219 ?

Comment (mateuszrzeszutek) [reply muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/ReferencesPrinter.java]: Done.

File: gradle-plugins/src/main/kotlin/io.opentelemetry.instrumentation.muzzle-check.gradle.kts [hunk 2]
```diff
@@ -336,3 +334,13 @@ fun muzzleDirectiveToArtifacts(muzzleDirective: MuzzleDirective, system: Reposit
 
   yieldAll(allVersionArtifacts)
 }
+
+fun withContextClassLoader(classLoader: ClassLoader, func: () -> Unit) {
```
Comment (iNikem) [inline gradle-plugins/src/main/kotlin/io.opentelemetry.instrumentation.muzzle-check.gradle.kts:338]: Rename to `action`?

**Review by iNikem — APPROVED** [review]

**Review by trask — APPROVED** [review]

**Pull request merged**